cmd/agent/main.go

```diff
@@ -1,27 +1,30 @@
-// Package main implements the root level command for the disruptor agent CLI
+// Package vli implements the root level command for the disruptor agent CLI
 package main
 
 import (
-	"context"
 	"fmt"
 	"os"
 
 	"github.com/grafana/xk6-disruptor/cmd/agent/commands"
+	"github.com/grafana/xk6-disruptor/pkg/agent"
 	"github.com/grafana/xk6-disruptor/pkg/runtime"
-	"github.com/spf13/cobra"
 )
 
 func main() {
 	env := runtime.DefaultEnvironment()
-	ctx := context.Background()
-
-	subcommands := []*cobra.Command{
-		commands.BuildHTTPCmd(env),
-		commands.BuildGrpcCmd(env),
+	config := &agent.Config{
+		Profiler: &runtime.ProfilerConfig{},
 	}
+	agent := agent.BuildAgent(env, config)
+
+	rootCmd := commands.BuildRootCmd(config)
+	rootCmd.AddCommand(commands.BuildHTTPCmd(agent))
+	rootCmd.AddCommand(commands.BuildGrpcCmd(agent))
```

Comment:
I'm not a huge fan of us creating configs this high in the call stack that then are passed down to different places at different levels.

A design that makes more sense in my mind is for the `rootCmd` to be a standalone wrapper of Cobra, and have it create and hold internally an instance of `agent.Config`. It could then create an `agent.Agent` lazily when a command is run.

This section would then look like:

```go
env := runtime.DefaultEnvironment()

rootCmds := commands.BuildRoot(env)
if err := rootCmds.Run() {
   //...
}
```

Where: 
- `commands.BuildRoot` creates an object holding the main cobra command and required config. It calls the equivalent of `commands.BuildHTTPCmd` internally, which could be private.
- `rootCmds.Run()` calls `rootCmd.Execute();` with `env.Args()[1:]`
- Individual `Run` function for commands instantiate an agent with their config.